src/main/webapp/app/exercises/programming/manage/update/update-components/custom-build-plans/programming-exercise-custom-aeolus-build-plan.component.ts

```diff
@@ -90,16 +91,19 @@ export class ProgrammingExerciseCustomAeolusBuildPlanComponent implements OnChan
         this.staticCodeAnalysisEnabled = this.programmingExercise.staticCodeAnalysisEnabled;
         this.sequentialTestRuns = this.programmingExercise.sequentialTestRuns;
         this.testwiseCoverageEnabled = this.programmingExercise.testwiseCoverageEnabled;
-        this.aeolusService
-            .getAeolusTemplateFile(this.programmingLanguage, this.projectType, this.staticCodeAnalysisEnabled, this.sequentialTestRuns, this.testwiseCoverageEnabled)
-            .subscribe({
-                next: (file) => {
-                    this.programmingExercise.windFile = this.aeolusService.parseWindFile(file);
-                },
-                error: () => {
-                    this.programmingExercise.windFile = undefined;
-                },
-            });
+        this.isImportFromFile = isImportFromFile;
+        if (!isImportFromFile || !this.programmingExercise.windfile) {
```

Comment:
Why is it not enough to check only `!this.programmingExercise.windfile`? From reading the code this sounds the most plausible here (if there is no configuration, load the default one, but if there is one keep it).

Reply:
I am not sure if there are cases where the windfile should be reloaded even though it was already defined, thus I added the `isImportFromFile` condition.